pkg/util/httpfilter/httpfilter.go

```diff
@@ -113,23 +114,35 @@ func (hf *HTTPFilter) Filter(ctx context.HTTPContext) bool {
 
 func (hf *HTTPFilter) filterHeader(ctx context.HTTPContext) bool {
 	h := ctx.Request().Header()
-	headerMatch := false
-	for key, vf := range hf.spec.Headers {
-		if headerMatch {
-			break
-		}
-
+	headerMatchNum := 0
```

Comment:
the change in this function impact performance.
Now, when `MatchAllHeaders` is false, the code will continue to match other headers when one header is matched.

Reply:
Updated with a pre-checking.